### DISCUSSION TITLE
Is there a way to create a reusable query function based on args

### DISCUSSION THREAD

**Discussion by SakoMe** [prisma/prisma#14500] [Q&A]
Is there a way to do something like this:

```
async function getData(tableName: string) {
  const data = await prisma[tableName].findMany({});
  await prisma.$disconnect();
  return data;
}
```

Then pass in a table name when invoking?

**Comment by austincrim** [comment]
Hey @SakoMe 👋,

You can do it exactly how you showed above using a dynamic property. Are you running into any issues using it that way?

**Reply by SakoMe** [reply]
Hi, thanks for the reply!  @austincrim TS is complaining: `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>'.
  No index signature with a parameter of type 'string' was found on type 'PrismaClient<PrismaClientOptions, never, RejectOnNotFound | RejectPerOperation | undefined>'.ts(7053)`

**Reply by austincrim** [reply]
This method is working for me with the following code:
```ts
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

async function getData(tableName: string) {
  const data = await prisma[tableName].findMany({})
  return data
}

async function main() {
  await getData('user')
}
```

Have you run `npx prisma generate`, and are you instantiating a new `PrismaClient`?

**Reply by SakoMe** [reply]
@austincrim Yes, I have but I have them in 2 separate files. 

```
import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient();

export default prisma;
```
And

```
import prisma from '../../../client';

async function getData(tableName: string) {
  const data = await prisma[tableName].findMany({});
  await prisma.$disconnect();
  return data;
}

export default getData;
```

And TS is complaining about `prisma[tableName]` with the message above

**Reply by SakoMe** [reply]
I can cast to `as any` and it will stop complaining:

```
async function getData(tableName: string) {
  const data = await (prisma as any)[tableName].findMany({});
  await prisma.$disconnect();
  return data;
}
```

**Reply by austincrim** [reply]
Hm, it still works for me even with that setup.

What version of Prisma are you using? 
What are the contents of your `tsconfig.json`?

**Reply by SakoMe** [reply]
prisma in dev dependencies prisma/client in dependencies version 4.0.0 and TS version 4.7.4

pretty standard tsconfig:

```
{
  "compilerOptions": {
    "target": "ESNext" 
    "module": "commonjs"
    "rootDir": "./src" 
    "outDir": "./dist"
    "esModuleInterop": true 
    "forceConsistentCasingInFileNames": true 
    "strict": true 
    "skipLibCheck": true 
  }
}
```

**Reply by SakoMe** [reply]
@austincrim Any suggestions? Seems there are a few more of these in the issue with minor variations but no answer to any.

**Reply by austincrim** [reply]
It looks like your `tsconfig.json` is slightly different, which is why it works for me without error.

I think it's probably best to just move forward with casting to `any` or a `ts-ignore` comment.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
